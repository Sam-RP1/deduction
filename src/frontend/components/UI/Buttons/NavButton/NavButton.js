import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

import '../Button.scss';

const NavButton = (props) => {
    return (
        <NavLink
            to={props.route}
            exact={true}
            onClick={() => {
                props.function();
            }}
        >
            <button className={'btn ' + props.opClasses}>{props.title}</button>
        </NavLink>
    );
};

NavButton.propTypes = {
    title: PropTypes.string.isRequired,
    route: PropTypes.string.isRequired,
    function: PropTypes.func.isRequired,
    opClasses: PropTypes.string,
};

NavButton.defaultProps = {
    title: 'UNDEFINED',
    route: '/undefined',
    function: () => console.log('[BUTTON] undefined button function'),
    opClasses: '',
};

export default NavButton;
