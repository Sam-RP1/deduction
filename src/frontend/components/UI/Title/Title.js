import React from 'react';
import PropTypes from 'prop-types';

import './Title.scss';

const Title = (props) => {
    return <h1 className='title'>{props.title}</h1>;
};

Title.propTypes = {
    title: PropTypes.string,
};

Title.defaultProps = {
    title: 'Deduction',
};

export default Title;
