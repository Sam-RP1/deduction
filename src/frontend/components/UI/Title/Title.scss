@import '../../../styles/partials/color-themes';
@import '../../../styles/mixins/transition';

.title {
  height: 90px;
  margin: 0;
  font-size: 5em;
  font-weight: 400;
  @include themed() { color: t($text); }
  @include transition-theme(color 0.4s ease);
  @media screen and (max-width: 350px) { font-size: 4.5em; }
}
