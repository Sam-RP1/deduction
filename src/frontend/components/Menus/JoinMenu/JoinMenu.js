import React from 'react';

import Title from '../../UI/Title/Title';
import NavButton from '../../UI/Buttons/NavButton/NavButton';

import './JoinMenu.scss';

const JoinMenu = () => {
    return (
        <section className='join-menu'>
            <Title title='Deduction' />

            <div className='join-menu__content'>
                <h3>Enter Join Link:</h3>
                <input />
            </div>

            <NavButton title={'Join!'} route={'/game'} />
        </section>
    );
};

export default JoinMenu;
