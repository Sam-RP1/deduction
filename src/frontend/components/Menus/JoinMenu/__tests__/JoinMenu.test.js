import React from 'react';

// Enzyme Imports
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

// Component Imports
import JoinMenu from '../JoinMenu';
import Title from '../../../UI/Title/Title';
import NavButton from '../../../UI/Buttons/NavButton/NavButton';

// Enzyme configuration
configure({ adapter: new Adapter() });

// Test cases
describe('<JoinMenu /> | Component', () => {
    it('should render <Title />', () => {
        const wrapper = shallow(<JoinMenu />);
        expect(wrapper.find(<Title />));
    });

    it('should render <NavButton />', () => {
        const wrapper = shallow(<JoinMenu />);
        expect(wrapper.find(<NavButton />));
    });

    it('should render <Title /> and <NavButton />', () => {
        const wrapper = shallow(<JoinMenu />);
        expect(wrapper.find(<Title />));
        expect(wrapper.find(<NavButton />));
    });

    it('should render correctly', () => {
        const wrapper = shallow(<JoinMenu />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
