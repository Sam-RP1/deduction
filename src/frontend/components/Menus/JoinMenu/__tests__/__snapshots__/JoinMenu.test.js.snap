// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JoinMenu /> | Component should render correctly 1`] = `
<section
  className="join-menu"
>
  <Title
    title="Deduction"
  />
  <div
    className="join-menu__content"
  >
    <h3>
      Enter Join Link:
    </h3>
    <input />
  </div>
  <NavButton
    function={[Function]}
    opClasses=""
    route="/game"
    title="Join!"
  />
</section>
`;
